generator/src/main/scala/HWRegPool.scala

```diff
@@ -84,12 +85,14 @@ trait PoolsMaster extends HWRegPool
   }
 }
 
-class XRegsPool extends ScalarRegPool
+class XRegsPool(loop: Boolean)  extends ScalarRegPool
 {
   val (name, regname, ldinst, stinst) = ("xreg", "reg_x", "ld", "sd")
   
   hwregs += new HWReg("x0", true, false)
-  for (i <- 1 to 31)
+  var last_reg = 31
+  if (loop) {last_reg = 30}
```

Comment:
I would prefer this to be done:

```
val last_reg = if (loop) 30 else 31
```

Then we don't need it to be a var as well.